pkg/apis/build/v1alpha1/trigger_when.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package v1alpha1
+
+// TriggerWhen a given scenario where the webhook trigger is applicable.
+type TriggerWhen struct {
+	// Name name or the short description of the trigger condition.
+	Name string `json:"name"`
+
+	// Type the event type
+	Type WhenTypeName `json:"type"`
```

Comment:
I would prefer to avoid the `When` in 
- WhenTypeName
- WhenGitHub
- WhenImage

as the `TriggerWhen` struct already have the `When`. 

Another disadvantage I see of this is on the comments of constants(see [example](https://github.com/otaviof/build/blob/shipwright-trigger-api/pkg/apis/build/v1alpha1/build_types.go#L67)), where the `TriggersInvalidGitHub indicates the when type GitHub is invalid` is hard to read, I would prefer something more like `TriggersInvalidGitHub indicates if the type GitHub is invalid`. wdyt? as this might need some changes on the API struct fields.

Reply:
Sure! I think it makes sense. I'm pushing changes with renamed variables and improved doc-comments. Please consider.